src/routes/user.js

```diff
@@ -0,0 +1,79 @@
+import express from 'express';
+import { viewProfile, updateProfile } from '../controllers/user';
+import profileSchema from '../validations/profileSchema';
+import middlewares from '../middlewares';
+
+const { validator, emptybody, asyncWrapper } = middlewares;
+const { profileUpdateSchema, profileViewSchema } = profileSchema;
+const router = express.Router();
+
+/**
+ * @swagger
+ *
+ * /profile/:id:
+ *   patch:
+ *     description: update user profile
+ *     produces:
+ *       - application/json
+ *     parameters:
+ *       - name: userName
+ *         description: Username to update.
+ *         in: body
+ *         required: true
+ *         type: string
+ *       - name: firstName
+ *         description: first name to update.
+ *         in: body
+ *         required: true
+ *         type: string
+ *       - name: lastName
+ *         description: last name to update.
+ *         in: body
+ *         required: true
+ *         type: string
+ *       - name: email
+ *         description: email to update.
+ *         in: body
+ *         required: true
+ *         type: string
+ *       - name: avaterUrl
+ *         description: image to update.
+ *         in: body
+ *         required: true
+ *         type: string
+ *     responses:
+ *       200:
+ *         description: updated successfully.
+ *       400:
+ *         description: not updated.
+ */
+router.patch(
+  '/profile/:id',
+  validator(profileUpdateSchema),
+  emptybody,
```

Comment:
Move the emptyBody middleware before the validator middleware. What's the need of doing the extra work when the body is empty anyways?